### PR TITLE
feat(learn): rename certificates, cert slugs, and cert flags

### LINKED ISSUE #38934: Confirm the shape of full-stack cert
The fullstack cert doesn't yet include the the three new python certificates, however it does include the InfoSec Cert and QA Cert (that have been split from the single InfoSec & QA cert).

We need clarity on the what is happening to the retired projects (there are new required projects in both certs).

We also need clarity on what is required to claim the fullstack cert. As is it does not have the three new python certs. At the same time, it needs one additional cert (from the split as mentioned earlier).

This could be a source of support queries. 

Refer: https://github.com/freeCodeCamp/freeCodeCamp/pull/38912#pullrequestreview-419094596

### LINKED ISSUE #39013: Rename Full Stack certification to Legacy Full Stack certification
Once https://github.com/freeCodeCamp/freeCodeCamp/pull/38995 is merged, the settings page will describe how to claim a Legacy Full Stack certification.

However, if you claim the Legacy Full Stack certification, it will show as a Full Stack certification instead. This is a reminder that we need to change this before the new [2020 Full Stack Development Certification](https://github.com/freeCodeCamp/freeCodeCamp/issues/38943) is created.

Related to https://github.com/freeCodeCamp/freeCodeCamp/issues/38934 and https://github.com/freeCodeCamp/freeCodeCamp/issues/38943.

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39013
Closes #38934

This PR renames the current Full Stack certificate to Legacy Full Stack. It also moves the requirements and claim/show button to the top of the Legacy Certifications section on the settings page.

### CODE REVIEW

**Comment by raisedadead** [comment]
A sanity check comes to mind:

We might have a problem with the slug (may not be related to the PR), we need to have a way to make sure the old full-stack slug still works.

Thanks.

**Comment by scissorsneedfoodtoo** [comment]
@raisedadead, thanks for pointing that out. I was wondering, is it better to handle the redirect for this slug change in `create-redirects.js`, or on the NGINX side of things? I can update this PR to include it, or submit another for the NGINX config.

**Comment by raisedadead** [comment]
The create-redirects file is no longer in use, it was something that we have had for generating the Netlify redirect files. We might as well clean it up and delete is.

We would need to leave the slug as is in this PR. The last thing we want is to be dependent on external logic (NGINX) for this. I think we would want to version control the slug by adding the year just like in the flags.

Let me think this over and get back to you.

**Comment by scissorsneedfoodtoo** [comment]
@raisedadead, that sounds good. Just let me know if there's anything else I can do in this PR or in another.

**Comment by raisedadead** [comment]
As discussed waiting on this to revert the slug back to `full-stack`

**Comment by scissorsneedfoodtoo** [comment]
@raisedadead, it's not so pretty, but this should revert all instances of `legacy-full-stack` to `full-stack`. 

I also realized that we should probably revert the`legacy-information-security-and-quality-assurance` slug to `information-security-and-quality-assurance` since those changes aren't live yet. I can do that in another PR the next chance I get.

**Comment by raisedadead** [comment]
> I also realized that we should probably revert the`legacy-information-security-and-quality-assurance` slug to `information-security-and-quality-assurance` since those changes aren't live yet. I can do that in another PR the next chance I get.

Good call. Looking forward to the PR.

File: curriculum/challenges/_meta/legacy-full-stack-certificate/meta.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "Full Stack Certificate",
-  "dashedName": "full-stack-certificate",
+  "name": "Legacy Full Stack Certificate",
+  "dashedName": "legacy-full-stack-certificate",
```
Comment (ojeytonwilliams) [inline curriculum/challenges/_meta/legacy-full-stack-certificate/meta.json:3]: I know we have some `legacy-x-certificates` already, but shouldn't this remain `full-stack-certificate` forever?  Only the text should be changed (like "name" and "title"), right?

Comment (raisedadead) [reply curriculum/challenges/_meta/legacy-full-stack-certificate/meta.json:3]: That is correct.

Comment (scissorsneedfoodtoo) [reply curriculum/challenges/_meta/legacy-full-stack-certificate/meta.json:3]: Yes, thanks for pointing that out. It doesn't seem to affect anything currently, but will make things easier to sort out in the future.

It's strange, but I played around with `"name"` and `"dashedName"` here. It seems like `"name"` controls the cert's `block` flag in the DB. Then `title` in `/curriculum/challenges/english/12-certificates/legacy-full-stack-certificate/legacy-full-stack-certificate.md` sets both `title` and `dashed-name` for the cert in the database.

That makes sense given how the other challenges are handled, though maybe something to keep in mind for generating a single source of truth for the certs client side.

**Comment by raisedadead** [comment]
Kris, I am closing the other umbrella issue in the OP as well. Let me know when this is ready for QA. No rush.

**Comment by scissorsneedfoodtoo** [comment]
@raisedadead, things are working now, but I had to change things all over the place. 

@ojeytonwilliams and I discussed having all the cert information in one place, and perhaps generated automatically from the markdown/`meta.json` files for each cert in the future.

Right now I'm just trying to refactor a bit and get all the cert/project information into one file manually.

**Comment by scissorsneedfoodtoo** [comment]
@ojeytonwilliams, I started the process of consolidating all the cert information into one file. Didn't have much of a chance to look at `TimeLine.js`, so please feel free to add to this if you have a chance. Otherwise I'll probably come back to it later this weekend.

Other than that, I tested all the certs locally and they seem to work. You can claim and view them all from the settings page, and all the links work on the public profile page.

**Review by ojeytonwilliams — COMMENTED** [review]
@scissorsneedfoodtoo everything seems fine to me.  I tried out the Timeline and it all seemed to work as expected.

Were you thinking of further refactoring, including that component?  If so, I'm on-board with that, but I think it would be best to hold off until after the launch and until we've got proper tests in place.

File: tools/scripts/build/__snapshots__/create-redirects.test.js.snap [hunk 2]
```diff
@@ -43,7 +43,6 @@ https://freecodecamp-org.netlify.com/*        https://www.freecodecamp.org/:spla
 /:username/front-end-certification                     /certification/:username/legacy-front-end 301
 /:username/data-visualization-certification            /certification/:username/legacy-data-visualization 301
 /:username/back-end-certification                      /certification/:username/legacy-back-end 301
-/:username/information-security-and-quality-assurance  /certification/:username/legacy-information-security-and-quality-assurance 301
```
[the remark this answers is not in the document]
Comment (raisedadead) [reply tools/scripts/build/__snapshots__/create-redirects.test.js.snap:46]: Lets remove this file and tests, we do not need these.

**Review by raisedadead — APPROVED** [review]
I am going to land this as is but https://github.com/freeCodeCamp/freeCodeCamp/issues/38609 would need some rethinking and hence blocker for overall deployment of the Python Curriculum

**Pull request merged**